### ISSUE TITLE
test failing on windows 10 WSL.

### ISSUE THREAD

**Issue by Aladdin-ADD** [eslint/eslint#8533]
**Tell us about your environment**

* **ESLint Version:**#master
* **Node Version:** v7.8.0
* **npm Version:** 4.5.0
* **os:** windows 10.0.16184 WSL

**`npm run test` output**

+ https://github.com/Aladdin-ADD/eslint/commit/0333d17d87a2d925700801c084a9fa9aa1947160

**Comment by mysticatea** [comment]
I'm using Windows 10 Pro, but I could not reproduce it.

![image](https://cloud.githubusercontent.com/assets/1937871/25685818/eeeaba44-30a4-11e7-9a0f-24fb7260eeb1.png)

**Comment by Aladdin-ADD** [comment]
actually I'm using Windows 10 WSL -- Windows Subsystem for Linux. sorry for a little misleading of the title.

I'm not sure it is a compatibility or just my computer's config problem. if need some more info, please let me know.

**Comment by soda0289** [comment]
@Aladdin-ADD I got two errors when running the tests on Bash for Windows 10.

```
  1) bin/eslint.js reading from stdin has exit code 0 if no linting errors are reported:

      Expected an exit code of 0 but got 1.
      + expected - actual

      -1
      +0

      at Function.assert.strictEqual (node_modules/chai/lib/chai/interface/assert.js:178:32)
      at awaitExit.then.exitCode (tests/bin/eslint.js:30:16)
      at process._tickCallback (internal/process/next_tick.js:109:7)

  2) bin/eslint.js reading from stdin gives a detailed error message if no config file is found:
     expected 'Implement me. Unknown stdin file type!\nError: Implement me. Unknown stdin file type!\n    at process.getStdin [as stdin] (internal/process/stdio.js:82:15)\n    at Object.<anonymous> (/mnt/c/Users/reyad.attiyat/Workspace/eslint/bin/eslint.js:58:12)\n    at Module._compile (module.js:570:32)\n    at Object.Module._extensions..js (module.js:579:10)\n    at Module.load (module.js:487:32)\n    at tryModuleLoad (module.js:446:12)\n    at Function.Module._load (module.js:438:3)\n    at Module.runMain (module.js:604:10)\n    at run (bootstrap_node.js:393:7)\n    at startup (bootstrap_node.js:150:9)\n' to match /ESLint couldn't find a configuration file/
```

The error message `Implement me. Unknown stdin file type!` I have seen before.
https://github.com/Microsoft/BashOnWindows/issues/1774

The problem was that STDIN was being closed before a child process could write to it. I only have Windows 10 at work but I will try and find some time to dig into this problem more.

**Comment by Aladdin-ADD** [comment]
oh.that's it. thanks for the link.

any suggestion? it blocked me a few days... 😿

**Comment by soda0289** [comment]
If you take a look at the test cases here:
https://github.com/eslint/eslint/blob/master/tests/bin/eslint.js#L63-L99

All the tests write to stdin then close the it. A hacky way to solve this would be to add a timeout around ` child.stdin.end();`. But to fix this properly we must wait for the child process to start up and then write to stdin and close it. Not sure how we can do this since there are only events for child_process exit. It might also be possible to use an event on the stdin stream itself but I'm not too sure.

**Comment by not-an-aardvark** [comment]
Does this depend on the event loop? If so, replacing the calls with `process.nextTick(() => child.stdin.end())` might work.

**Comment by soda0289** [comment]
@not-an-aardvark I do not know enough about node internals to say if that will work but it is better than a random timeout. 

@Aladdin-ADD give the `nextTick()` callback a shot and see if it fixes the problem

**Comment by Aladdin-ADD** [comment]
it works, but output other errors as before. 
>CLIEngine "before all" hook:
>     Error: Timeout of 10000ms exceeded. For async tests and hooks, ensure "done()" is called; if returning a Promise, ensure it resolves.

I will try to build the latest nodejs to see if it fixed. 

@soda0289 ,looking forward to your better way.

**Comment by soda0289** [comment]
I was not able to reproduce those errors but I only tested it on node v6.10.2 and windows 10 build 15063. Will test out node v7.10.

**Comment by not-an-aardvark** [comment]
Is this still a problem?

**Comment by Aladdin-ADD** [comment]
yes. I just tested, and got an error.

```bash
  1) globUtil listFilesToProcess() should not return hidden files for standard glob patterns:

      AssertionError: expected 1 to equal 0
      + expected - actual

      -1
      +0

      at Context.it (tests\lib\util\glob-util.js:216:20)
```

**Comment by Aladdin-ADD** [comment]
closing, as it is not eslint's bug.
